tests/blackbox/bot_challenge/test_bb_bot_challenge.py

```diff
@@ -364,3 +362,45 @@ def test_bot_challenge_with_profile(setup_scopez_server_action):
     l_parser.feed(l_r.text)
     assert 'function' in l_parser.m_data
 
+# ------------------------------------------------------------------------------
+# test /update_bots endpoint
+# ------------------------------------------------------------------------------
+def test_update_bots_endpoint(setup_scopez_server_action):
+    # ------------------------------------------------------
+    # test for recieving a bot challenge with attack vector
+    # ------------------------------------------------------
+    l_url = G_TEST_HOST + '/update_bots'
+
+    l_file_path = os.path.dirname(os.path.abspath(__file__))
+    l_test_file = os.path.realpath(os.path.join(l_file_path,
+                                                '../../data/waf/conf/bots/0052-wHyMHxV7.bots.json'))
+    l_test_payload = ''
+
+    # check setup
+    assert os.path.exists(l_test_file), 'test file not found!'
+
+    # slurp test file
+    with open(l_test_file) as l_tf:
+        l_test_payload = l_tf.read()
+
+    # check setup
+    assert l_test_payload, 'payload is empty!'
+
+    l_json_payload = json.loads(l_test_payload)
+
+    # check positive test
+    l_result = requests.post(l_url, timeout=3, json=l_json_payload)
+    assert l_result.status_code == 200
+    assert l_result.json()['status'] == 'success'
+
+
+    # check negative test - test id does not match scopes map
+    l_json_payload['id'] = '1d0ntex15t'
+    l_n1_result = requests.post(l_url, json=l_json_payload)
+    assert l_n1_result.status_code == 500
+
+    # check negative test - missing customer_id field
+    l_cust_id = l_json_payload.pop('customer_id')
+    l_n2_result = requests.post(l_url, json=l_json_payload)
+    assert l_n2_result.status_code == 500,\
+        'expected 500 since customer_id {} is removed'.format(l_cust_id)
```

Comment:
i think this test should move to blackbox/scopes/test_bb_scopes_updates.py, thats where all the tests for config updates currently resides.. this is good negative test, but you need to add functional case to it which verifies bot config is successfully updated in memory. you can refer "test_rules_config_update" and other entity updates.. 
It can be like this.
1. Make a request with host/path for the particular scope in which bot config(0052-wHyMHxV7) is defined. You should also add corresponding user-agents defined in the custom rule in 0052-wHyMHxV7.bots.json so that bots get triggered and should get a browser challenge in response.
2. Verify you get a browser challenge in response.
3. sleep for browser challenge to expire.
4. change something in the content of bot config (0052-wHyMHxV7.bots.json) like different user agent .
5.  update bot config
6. if you make a request with old user agent , you shouldn't get browser challenge as enforcement
7. But if you make a request with new user agent, you should get browser challenge as enforcement 
This will make sure things are updated properly in memory

Reply:
updated tests, thanks